src/Collections/MemoryConfigCollection.php

```diff
@@ -215,6 +215,10 @@ public function getAll()
      */
     public function update($class, $name, $value)
     {
+        user_error(
+            'Config::inst()->update() is deprecated. Please use YAML configuration, Config::modify()->merge() or ->set()',
+            E_USER_WARNING
```

Comment:
Actually, you get this error even if you call Config::modify()->update(), so you aren't going to capture the deprecated behaviour with this change.

If you never call ::modify() the instance will be a CachedConfigCollection which does NOT implement the MutableConfigCollectionInterface.